pythclient/price_feeds.py

```diff
@@ -0,0 +1,430 @@
+        price_attestations.append(
+            parse_price_attestation(bytes_[offset : offset + attestation_size])
+        )
+        offset += attestation_size
+
+    if offset != len(bytes_):
+        raise ValueError(f"Invalid length: {len(bytes_)}, expected: {offset}")
+
+    return {
+        "price_attestations": price_attestations,
+    }
+
+
+def parse_price_attestation(bytes_):
+    offset = 0
+
+    product_id = bytes_[offset : offset + 32].hex()
+    offset += 32
+
+    price_id = bytes_[offset : offset + 32].hex()
+    offset += 32
+
+    price = int.from_bytes(bytes_[offset : offset + 8], byteorder="big", signed=True)
+    offset += 8
+
+    conf = int.from_bytes(bytes_[offset : offset + 8], byteorder="big", signed=False)
+    offset += 8
+
+    expo = int.from_bytes(bytes_[offset : offset + 4], byteorder="big", signed=True)
+    offset += 4
+
+    ema_price = int.from_bytes(
+        bytes_[offset : offset + 8], byteorder="big", signed=True
+    )
+    offset += 8
+
+    ema_conf = int.from_bytes(
+        bytes_[offset : offset + 8], byteorder="big", signed=False
+    )
+    offset += 8
+
+    status = int.from_bytes(bytes_[offset : offset + 1], byteorder="big")
+    offset += 1
+
+    num_publishers = int.from_bytes(
+        bytes_[offset : offset + 4], byteorder="big", signed=False
+    )
+    offset += 4
+
+    max_num_publishers = int.from_bytes(
+        bytes_[offset : offset + 4], byteorder="big", signed=False
+    )
+    offset += 4
+
+    attestation_time = int.from_bytes(
+        bytes_[offset : offset + 8], byteorder="big", signed=True
+    )
+    offset += 8
+
+    publish_time = int.from_bytes(
+        bytes_[offset : offset + 8], byteorder="big", signed=True
+    )
+    offset += 8
+
+    prev_publish_time = int.from_bytes(
+        bytes_[offset : offset + 8], byteorder="big", signed=True
+    )
+    offset += 8
+
+    prev_price = int.from_bytes(
+        bytes_[offset : offset + 8], byteorder="big", signed=True
+    )
+    offset += 8
+
+    prev_conf = int.from_bytes(
+        bytes_[offset : offset + 8], byteorder="big", signed=False
+    )
+    offset += 8
+
+    last_attested_publish_time = int.from_bytes(
+        bytes_[offset : offset + 8], byteorder="big", signed=True
+    )
+    offset += 8
+
+    return {
+        "product_id": product_id,
+        "price_id": price_id,
+        "price": str(price),
+        "conf": str(conf),
+        "expo": expo,
+        "ema_price": str(ema_price),
+        "ema_conf": str(ema_conf),
+        "status": status,
+        "num_publishers": num_publishers,
+        "max_num_publishers": max_num_publishers,
+        "attestation_time": attestation_time,
+        "publish_time": publish_time,
+        "prev_publish_time": prev_publish_time,
+        "prev_price": str(prev_price),
+        "prev_conf": str(prev_conf),
+        "last_attested_publish_time": last_attested_publish_time,
+    }
+
+
+def vaa_to_price_info(price_feed_id, vaa) -> PriceInfo:
```

Comment:
making classes per the comment below would actually help you separate out the logic into composable bits.

Reply:
one more note here for the future: we need to come up with a term for "a binary blob of data that you can post on-chain to update the pyth price". We are currently calling these "vaas" but I don't think that's right anymore in the accumulator world.

Reply:
do you mean creating a separate function like `vaa_to_price_infos` which returns a `List[PriceInfo]` and then make `vaa_to_price_info` call `vaa_to_price_infos` and return the `PriceInfo` from the list where it has the specified `id`?

Reply:
it depends on how often people want to use `vaa_to_price_info`. If that's a common function that people want, then yes, you can have it. Otherwise, people can call `find` on the list returned by `vaa_to_price_infos`.